### DISCUSSION TITLE
setting initial conditions

### DISCUSSION THREAD

**Discussion by francispoulin** [CliMA/Oceananigans.jl#3781] [Computational science!]
I am currently using `OceanBioME` to test different models of biogeochemistry.  

I set set things up to loop over the different models and have been able to do everything clearly except for the following,

```
    if i==1  #FJP: fix this!
        set!(model, P = 0.03, Z = 0.03, NO₃ = 4.0, NH₄ = 0.05);
    elseif i==2
        set!(model, P = 0.03, Z = 0.03, N = 4.0, T = 10);
    end
```

Is there a way I can define a dictionary (or something else) and store the initial conditions in there and then specify that in set?  

The problem is that there are different fields for the two cases so the initial conditions must be different.

**Comment by simone-silvestri** [comment]
do you mean something like this?
```julia
initial_conditions = Dict(
    1 => (; a = 1, b = 2, c = 3),
    2 => (; a = 4, b = 5, d = 3),
)

function test(; a = 1, b = 2, c = 3, d = 4)
    @show a, b, c, d
end
```
where 
```julia
julia> test(; initial_conditions[2]...)
(a, b, c, d) = (4, 5, 3, 3)
(4, 5, 3, 3)

julia> test(; initial_conditions[1]...)
(a, b, c, d) = (1, 2, 3, 4)
(1, 2, 3, 4)
```

In this way, you could write up all your cases in the one dictionary
```julia
initial_conditions = Dict(
    1 => (; P = 0.03, Z = 0.03, NO₃ = 4.0, NH₄ = 0.05),
    2 => (; P = 0.03, Z = 0.03, N = 4.0, T = 10),
)
```
and then 
```julia
for i in eachindex(models)
     set!(models[i]; initial_conditions[i]...)
end
```

would save you the if condition but not much more...

**Reply by francispoulin** [reply]
Yes, that will definitely do it.  Thanks for the clever idea and quick reply!
